src/components/GuppyWrapper/index.jsx

```diff
@@ -174,21 +178,21 @@ class GuppyWrapper extends React.Component {
    * This function uses current filter argument
    */
   handleDownloadRawDataByFields({ fields, sort = [] }) {
-    let targetFields = fields;
-    if (typeof fields === 'undefined') {
-      targetFields = this.state.rawDataFields;
-    }
-    return downloadDataFromGuppy(
+    return askGuppyForTotalCounts(
+      this.props.guppyConfig.path,
+      this.props.guppyConfig.type,
+      this.state.filter,
+      'accessible',
+    ).then((count) => downloadDataFromGuppy(
       this.props.guppyConfig.path,
       this.props.guppyConfig.type,
-      this.state.totalCount,
+      count,
       {
-        fields: targetFields,
+        fields: fields || this.state.rawDataFields,
```

Comment:
The `this.state.rawDataFields` value is the one coming from the gitops.json/pcdc.json or the field on is?

Reply:
It's based on the array of fields for table in configuration file--`dataExplorerConfig.table.fields`. If not provided, it uses all fields.